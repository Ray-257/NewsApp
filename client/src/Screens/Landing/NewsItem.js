import React from "react";

const NewsItem = ({ item }) => {
  const formatDate = (s) =>
    new Date(s).toLocaleDateString(undefined, { dateStyle: "long" });
  return (
    <li className="item">
      <h2 className="title">
        <a href={item.url}>{item.title}</a>
      </h2>
      <p className="description">{item.description || ""}</p>
      <div className="meta">
        <span>{formatDate(item.publishedAt)}</span>
      </div>
    </li>
  );
};

export default NewsItem;
