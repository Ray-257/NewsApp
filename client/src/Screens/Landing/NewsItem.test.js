import React from 'react';
import { shallow } from 'enzyme';
import NewsItem  from './NewsItem';


describe('NewsItem', () => {
  let wrapper

  beforeEach(() => {
     wrapper = shallow(<NewsItem  item={{title:'News title 1',description:"Testing news post"}}/>);
  })
  afterEach(() => {
    wrapper = null
  })
  it('should render correctly', () => {
    expect(wrapper).toMatchSnapshot();
  });
  it('should have  title', () => {
    const title = wrapper.find("a").text();
    expect(title).toEqual("News title 1");
  });
  it('should have description', () => {
    const title = wrapper.find("p").text();
    expect(title).toEqual("Testing news post");
  });
});