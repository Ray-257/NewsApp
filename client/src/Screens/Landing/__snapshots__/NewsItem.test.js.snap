// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsItem should render correctly 1`] = `ShallowWrapper {}`;
